cobalt/base/process/process_metrics_helper.cc

```diff
@@ -178,4 +179,60 @@ TimeDelta ProcessMetricsHelper::GetCPUUsage(const FilePath& path,
                                            ticks_per_s);
 }
 
+// static
+int ProcessMetricsHelper::GetNumberOfConfiguredProcessors(
+    ReadCallback read_callback) {
+  absl::optional<std::string> contents = std::move(read_callback).Run();
+  if (!contents.has_value()) {
+    // Expected for non-Linux platforms.
+    LOG(WARNING) << "Not supported for platform.";
+    return -1;
+  }
+
+  auto chunks =
+      SplitString(*contents, ",", TRIM_WHITESPACE, SPLIT_WANT_NONEMPTY);
+  int count = 0;
+  int lowerProcessorNumber = -1;
+  int upperProcessorNumber = -1;
+  for (const auto& chunk : chunks) {
+    auto chunkParts =
+        SplitString(chunk, "-", TRIM_WHITESPACE, SPLIT_WANT_NONEMPTY);
+    if (chunkParts.size() == 0 || chunkParts.size() > 2) {
+      LOG(ERROR) << "Unexpected format of processor count chunks.";
+      return -2;
+    }
+    if (!StringToInt(chunkParts[0], &lowerProcessorNumber)) {
+      LOG(ERROR) << "Unexpected format of processor count chunks.";
+      return -3;
+    }
+    if (chunkParts.size() == 1) {
+      count++;
+      continue;
+    }
+    if (!StringToInt(chunkParts[1], &upperProcessorNumber)) {
+      LOG(ERROR) << "Unexpected format of processor count chunks.";
+      return -3;
+    }
+    if (upperProcessorNumber <= lowerProcessorNumber) {
+      LOG(ERROR) << "Unexpected format of processor count chunks.";
+      return -4;
+    }
+    count += 1 + upperProcessorNumber - lowerProcessorNumber;
+  }
+
+  return count;
+}
+
+// static
+int ProcessMetricsHelper::GetNumberOfConfiguredProcessors() {
+  int stored_value = number_of_configured_processors.load();
+  if (stored_value != 0) {
+    return stored_value;
+  }
+  int result = GetNumberOfConfiguredProcessors(
+      GetReadCallback(FilePath("/sys/devices/system/cpu/possible")));
```

Comment:
Did you have a look if using `SbCPUFeatures` would be more robust here ? https://developers.google.com/youtube/cobalt/docs/reference/starboard/modules/cpu_features